packages/contracts/src/OffchainResolver.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./IExtendedResolver.sol";
+import "./SignatureVerifier.sol";
+
+/**
+ * Implements an ENS resolver that directs all queries to a CCIP read gateway.
+ * Callers must implement EIP 3668 and ENSIP 10.
+ */
+contract OffchainResolver is IExtendedResolver, IERC165, Ownable {
+    string public url;
+    mapping(address => bool) public signers;
+
+    event NewSigners(address indexed signer, bool isSigner);
+    event UpdateUrl(string url);
+
+    error OffchainLookup(
+        address sender,
+        string[] urls,
+        bytes callData,
+        bytes4 callbackFunction,
+        bytes extraData
+    );
+
+    constructor(string memory _url, address[] memory _signers) {
+        url = _url;
+        emit UpdateUrl(_url);
+
+        uint256 arrayLength = _signers.length;
+        for (uint256 i; i < arrayLength; ) {
+            signers[_signers[i]] = true;
+            emit NewSigners(_signers[i], true);
+
+            unchecked {
```

Comment:
what does this means?

Reply:
Basically, when an event is emitted, it creates a log entry in the blockchain. These logs can be filtered and searched by external applications, allowing them to monitor the behaviour of the smart contract. So, create this emits facilitates to other parts of the project to ensure that every thing is working well in the smart contracts.
In our project's scope, it may not be useful since we still dont have a handler for this emits. Do you prefer to remove it, and just add again if the necessity appears?

Reply:
I mean the `unchecked`

Reply:
Oh, i see. The uncheck is for gas economy since the operation i++ will never over/underflow. The "i" variable will not be greater then arrayLength that is already checked.